packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -1025,6 +1003,29 @@ export class ThinEngine {
         }
     }
 
+    protected _setupMobileChecks(): void {
+        if (!(navigator && navigator.userAgent)) {
+            return;
+        }
+
+        // Function to check if running on mobile device
+        this._checkForMobile = () => {
+            const currentUA = navigator.userAgent;
+            this.hostInformation.isMobile =
+                currentUA.indexOf("Mobile") !== -1 ||
+                // Needed for iOS 13+ detection on iPad (inspired by solution from https://stackoverflow.com/questions/9038625/detect-if-device-is-ios)
+                (currentUA.indexOf("Mac") !== -1 && IsDocumentAvailable() && "ontouchend" in document);
+        };
+
+        // Set initial isMobile value
+        this._checkForMobile();
+
+        // Set up event listener to check when window is resized (used to get emulator activation to work properly)

```

Comment:
I assume this is the only use case for this callback? Why doesn't it work the first time it is executed?

Reply:
Do you mean, why do we have to call it in the window resize event? If that is so, there's this comment:
``` 
Set up event listener to check when window is resized (used to get emulator activation to work properly)
``` 
Note that I didn't write this code, I just moved it in a separate method.

Reply:
yeah, this is what i was referring too. I saw the comment, was just wondering if it is really needed. We can check that at a later point